modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/Swift5ClientCodegen.java

```diff
@@ -342,7 +342,15 @@ protected void addAdditionPropertiesToCodeGenModel(CodegenModel codegenModel,
         final Schema additionalProperties = getAdditionalProperties(schema);
 
         if (additionalProperties != null) {
-            codegenModel.additionalPropertiesType = getSchemaType(additionalProperties);
+            Schema inner = null;
+            if (ModelUtils.isArraySchema(schema)) {
```

Comment:
@aymanbagabas could you please explain what this change does?

Reply:
This was getting the wrong inner type for `additionalProperties` when the schema is either an array or a map.

Reply:
Thanks for explaining the change 🙂